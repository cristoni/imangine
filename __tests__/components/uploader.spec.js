import React from 'react';
import Uploader from '../../components/uploader';
import Dropzone from 'react-dropzone';

test('Index renders the main page', () => {
  const wrapper = mount(<Uploader />);
  expect(wrapper).toMatchSnapshot();

  // const fileContents = "file contents";
  // const fpng = new Blob([fileContents], { type: "image/png" });
  // const fjpg = new Blob([fileContents], { type: "image/jpeg" });
  // const ftxt = new Blob([fileContents], { type: "text/plain" });
  console.log(wrapper.find('.dropzone'));
  expect(wrapper.find('.dropzone')).to.have.lengthOf(1);
  // wrapper.find(Dropzone).simulate("drop", { dataTransfer: { files: [fpng, fjpg, ftxt] } });
});