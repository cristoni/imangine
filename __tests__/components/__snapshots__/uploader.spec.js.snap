// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders the main page 1`] = `
<Uploader>
  <n
    accept="image/jpeg, image/png"
    activeClassName="dropzone--active"
    className="dropzone d-flex justify-content-center align-items-center shadow"
    disableClick={false}
    disabled={false}
    getDataTransferItems={[Function]}
    inputProps={Object {}}
    maxSize={Infinity}
    minSize={0}
    multiple={true}
    preventDropOnDocument={true}
  >
    <div
      aria-disabled={false}
      className="dropzone d-flex justify-content-center align-items-center shadow"
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <h6
        className="text-uppercase text-center"
      >
        Drop your files here or click to open the file explorer
      </h6>
      <input
        accept="image/jpeg, image/png"
        autoComplete="off"
        disabled={false}
        multiple={true}
        onChange={[Function]}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0.00001,
            "pointerEvents": "none",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        type="file"
      />
    </div>
  </n>
</Uploader>
`;
